### ISSUE TITLE
[q] How to add messages to stream

### ISSUE THREAD

**Issue by SkeLLLa** [nodefluent/kafka-streams#50]
Hello. Brief problem description - at some point I receive a message. Then it's processed and as a result, I need to push two messages into output topic. What is the best way to do that?

```
eventStream
        .from('intopic')
        .map(message => {
           // do something
           // send two messages into the same topic
        })
        .to('outtopic')
```

I've tried to use writeMessage, but it adds the message to the start of the stream.
Probably there's a way to directly call producer send method?
Or maybe it's better to create a standalone producer (or stream) and write additional message into in and after `.map` merge/concat it with existing stream?

**Comment by krystianity** [comment]
Hi @SkeLLLa you are right, _writeToStream_ will always add to the beginning of the internal most stream. 

Accessing the raw Kafka client is not too difficult, but not very pretty either. It is part of the stream though, you can reach it by "eventStream.kafka.x", calling produce functions is a bit more complicated (because of different optional configuration settings) [used call overview](https://github.com/nodefluent/kafka-streams/blob/master/lib/messageProduceHandle.js#L50)

Creating a second stream without read-topic, that you merge and use only to produce to, is a decent way to solve this use-case :) However you might need to be carefull with call order.

**Comment by SkeLLLa** [comment]
Thanks for an answer.

However, I've spent some time playing around with most and have another idea.
```js
const most = require('most');
stream
  .from('input_topic')
  .mapJSONConvenience()
  .concatMap((msg) => {
    return most.from([
      Promise.resolve({taskId: 1, type: 1, value: msg.value}),
      Promise.resolve({taskId: 1, type: 2, value: 'type_2'}),
    ]);
  })
  .awaitPromises()
  .map((message) => {
    return {
      payload: message,
      time: new Date().toISOString(),
      type: message.type,
      id: message.taskId,
    };
  })
  .to('out_topic', 1, 'buffer');
```

@krystianity what do you think about this one? If it's ok, then I'll add new `concatMap` method from most js to StreamSDL.js

**Comment by krystianity** [comment]
This solution is even better then a second stream :)

I would like to wrap the creation of a second most stream though, just to make sure we can replace the underlying stream lib at any time in the future without a breaking change.
Feel free to create a PR 👍

**Comment by krystianity** [comment]
Hi @SkeLLLa sry it took me a while to find time - I have merged this and released @4.3.0, additionally I have added the internal most function as well as new example to show you what I meant (and in case someone else has a similiar questions (as last issue) in the future).
You can find that [here](https://github.com/nodefluent/kafka-streams/blob/master/examples/consumeOneProduceTwo.js#L13)
